addie/calculate_gr/event_handler.py

```diff
@@ -206,17 +209,71 @@ def generate_gr_step2(main_window, sq_ws_name_list):
 
     # loop for all selected S(Q)
     for sq_ws_name in sq_ws_name_list:
-        # calculate G(r)
-        gr_ws_name = main_window._myController.calculate_gr(
-            sq_ws_name,
-            pdf_type,
-            min_r,
-            delta_r,
-            max_r,
-            min_q,
-            max_q,
-            pdf_filter,
-            rho0)
+        if main_window.calculategr_ui.ff_check.checkState() == 2:
+            if rho0 is None:
+                print("WARNING: rho0 is not a float. Necessary for applying meaningful Fourier filter.")
+                return
+            # calculate G(r)
+            gr_ws_name_temp = main_window._myController.calculate_gr(
+                sq_ws_name,
+                pdf_type,
+                min_r,
+                delta_r,
+                max_r,
+                min_q,
+                max_q,
+                pdf_filter,
+                rho0)
+            # Fourier filter
+            out_ws_temp = AnalysisDataService.retrieve(sq_ws_name)
+            out_ws_r_temp = AnalysisDataService.retrieve(gr_ws_name_temp)
```

Comment:
change from `gr_ws_name` -> `gr_ws_name`

Reply:
Not sure what you mean by this comment.

Reply:
Ha, sorry, meant change from `gr_ws_name_temp` -> `gr_ws_name`.
Basically, if you address the comments above and below, you can remove the `_temp` references.

Reply:
I see. Doing the change now. It will be very nice of you if you can help with the `pystog` part. is that OK for you?